Upgrade cs lib

greg0ire opened the pull request.

greg0ire committed:
Upgrade cs lib

greg0ire committed:
Display process output on failure

greg0ire committed:
Use a more recent (and less buggy?) version of box

3.1.0 crashes with syntax error, unexpected 'fn' (T_FN), expecting
  identifier (T_STRING)

greg0ire committed:
Update some of the packages

greg0ire committed:
Do not use /tmp as a migration directory

You do not know what it may contain. If it contains directories
unreadable for the current user, for instance, the test suite will crash
because these cannot be opened.

greg0ire commented:
The Travis build fails because of an issue I reported here: https://github.com/box-project/box/issues/489

smoench commented on .github/workflows/continuous-integration.yml:
```diff
@@ -0,0 +1,98 @@
+      - "phpunit"
+
+    steps:
+      - name: "Checkout"
```
Have you tried with the job without the checkout step? I think there is no need for and will saves build time.

greg0ire replied:
Oh you're right, that should work indeed.

smoench commented on .github/workflows/continuous-integration.yml:
```diff
@@ -0,0 +1,98 @@
+      - name: "Run PHPUnit"
+        run: "vendor/bin/phpunit --coverage-clover=coverage.xml"
+
+      - name: "Upload coverage file"
```
I would recommend uploading the coverage file only once and enable the coverage run only on a specific php version. It will saves build time. Here an example how it could be done https://github.com/humbug/php-scoper/pull/422

greg0ire replied:
What if I have code that only execute on a specific version of PHP?

goetas replied:
is Codecov able to distinguish them? (i remember that scrutinizer was considering only the first received file...)

greg0ire replied:
Yes, it can tell you how many jobs cover a single line

smoench replied:
Nevermind, now unterstand what the coverage steps are doing. Looks fine!

greg0ire replied:
It's a bit more complex that uploading directly to Codecov, I had to do it on the DBAL where those upload would fail because of a limit in the number of jobs, and ending up doing this: one upload of all artifacts in one step. It's a bit more complicated but I think it might make the build more stable by reducing the possibility of a failure when uploading to Codecov.

goetas commented on tests/Doctrine/Migrations/Tests/Tools/Console/Command/MigrationVersionTest.php:
```diff
@@ -33,7 +36,11 @@ protected function setUp() : void
         $this->configuration = new Configuration($this->getSqliteConnection());
         $this->configuration->setMigrationsNamespace('DoctrineMigrations');
-        $this->configuration->setMigrationsDirectory(sys_get_temp_dir());
+        $migrationsDirectory = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'doctrine-migrations';
```
good catch!

greg0ire replied:
Yup, that one gave a weird error that was hard to understand :smile:

greg0ire committed:
Temporarily skip test

A bug in box prevents it from running on PHP 7.1

greg0ire commented:
Blocked by https://github.com/box-project/box/issues/490

goetas reviewed (approved):
I am good with this changes. They improve the current state of the codebase.

goetas commented on download-box.sh:
```diff
@@ -1,5 +1,5 @@
 
 if [ ! -f box.phar ]; then
-    wget https://github.com/humbug/box/releases/download/3.1.0/box.phar -O box.phar
+    wget https://github.com/box-project/box/releases/download/3.9.0/box.phar -O box.phar
```
this is giving 404

greg0ire replied:
Yeah, I'm not sure why I committed that, anyway we probably won't use 3.9.0 either since it still suffers from the version detection bug. [No need to report the issue though, the team is aware of it](https://github.com/box-project/box/issues/490)

goetas reviewed (dismissed):
there is a 404 to download the box

greg0ire commented:
Converting to draft until the numerous issues with box are resolved.

greg0ire committed:
Fix PHPStan findings

greg0ire commented:
This is too complex, I will move to GA in a separate PR.

greg0ire committed:
Use stable image for PHP 7.4

greg0ire committed:
Always test PHAR compilation

greg0ire commented:
I just noticed that the PHAR compilation was only ever tested with low dependencies. With 96ce53a, it's not tested at all, and I don't understand if testing it only once was deliberate. @jwage , maybe you remember? I believe you changed this in #700

greg0ire commented:
And of course now it fails with the same errors I had when attempting to migrate to GA… that had nothing to do with GA.

greg0ire commented:
Good news: I'm reproducing the failure locally. Also, I reproduce the failure without PHPUnit if I run the compilation from the `tests` directory. I think the current working directory might be important.

greg0ire commented:
Ah apparently this is supposed to be fixed by 3.9.0: https://github.com/box-project/box/commit/e65178c5edaf5012b8b6ae9137274f76a4b0a572 … which cannot be downloaded as a PHAR : https://github.com/box-project/box/issues/490

goetas commented:
> Ah apparently this is supposed to be fixed by 3.9.0: box-project/box@e65178c … which cannot be downloaded as a PHAR : box-project/box#490
:cry:

greg0ire commented:
A version of the PHAR was just manually uploaded! It fixes the build locally, I just pushed :tada:

greg0ire committed:
Apply new dependencies policy

We no longer put composer.lock under version control and resort to
composer install. Instead, if a tool breaks often, we use tighter
version constraints for it.

goetas reviewed (approved):
:+1:  looks good!
